xtask/Cargo.toml

```diff
@@ -19,4 +19,6 @@ ungrammar = "1.1.3"
 walkdir = "2.3.1"
 write-json =  "0.1.0"
 fs-err = "2.3"
+serde = { version = "1.0", features = ["derive"] }
+serde_json = "1.0"
```

Comment:
I don't think adding serde to xtask is a good tradeoff. Quantitatively, this makes xtask 3x longer to compile in debug mode. 

Not sure what's the best way to solve this. It seems that code-generating a big blog of rust structs might be a problem in itself. Fundamentally, we are dealing with a data table here, serializing it to code seems wrong.

Perhaps, instead of generating huge blobs of code, we should generate a `.json` file, describing available lints, `include_str!` it, and deserialize it at runtime? But then, again, I think the code of rust-analyzer doesn't depend on serde, and that's valuable for scope management. Perhaps we should dump info into a really simple text file, which can be parsed using `.ilnes` and `.split`, and load that at runtime?

Reply:
If that Rust PR gets merged, we can invoke `rustc` on start-up?

Reply:
Yeah, that also would work! I wonder if we can use similar runtime discovery for feature flags?

Reply:
Maybe we can `include_str!` the lints in the same format as rustc would output them for now, then we can just keep the parsing code when we switch to runtime discovery later?

Reply:
I just pushed an update, in fact I just deleted the serde dependency and parse the lint.json file by myself. Let me know :)

Reply:
> Yeah, that also would work! I wonder if we can use similar runtime discovery for feature flags?

Filed https://github.com/rust-lang/rust/issues/77937